docs/parallel-mode.md

```diff
@@ -0,0 +1,76 @@
+# Parallel Pull and Unpack Mode
+
+## Introduction
+
+The SOCI Snapshotter's lazy loading technology has proven effective in accelerating container startup times. The technology defers the full image download and unpacking until the container actually needs the files, allowing for near-instant container launches.
+
+However, some users prefer the traditional upfront loading model, where the entire container image is pulled and prepared before the container is started. This is particularly the case for I/O-bound applications that require fast access to the full rootfs.
+
+To support users who favor the upfront loading workflow, we have introduced a new _parallel-pull-unpack_ mode in the SOCI Snapshotter. This mode enables concurrent processing of container image layers, significantly improving the performance of the upfront loading process.
+
+## Functionality
+ 
+The _parallel-pull-unpack_ mode operates as follows:
+
+* Parallel Layer Downloads: When preparing a container image snapshot, the snapshotter initiates concurrent HTTP range GET requests to fetch the individual image layers. Each layer is downloaded in parallel using multiple HTTP connections, with the downloaded data being written to a temporary file on disk rather than buffered in memory.
+
+* Parallel Layer Unpacking: After the layers are downloaded, the snapshotter also launches parallel unpacking operations. Multiple layers can be decompressed and extracted concurrently.
+
+## Use Cases
+
+The _parallel-pull-unpack_ mode is most beneficial in the following scenarios:
+
+* Complex, Multi-Layer Images: Applications that rely on container images with a large number of layers, such as machine learning frameworks or language runtimes, stand to gain the most from the parallel processing capabilities.
```

Comment:
nit: I think this and the `performance tuning` section would be easier to read if the title of the bullet was bold.